Anlab.Mvc/Controllers/LabController.cs

```diff
@@ -826,6 +827,79 @@ public async Task<ActionResult> Search(string term)
 
         }
 
+        public async Task<IActionResult> DisposalList()
+        {
+            var disOrders = await _dbContext.DisposalView.Where(a => a.DateFinalized.Value.Date >= DateTime.UtcNow.Date.AddDays(-31)).AsNoTracking().ToListAsync();
+            Message = "Showing Disposal Orders Finalized within ~31 days";
+            return View(disOrders);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> GenerateDisposalEmail(int id)
+        {
+            var order = await _dbContext.Orders.Include(i => i.Creator).SingleOrDefaultAsync(o => o.Id == id);
+
+            if (order == null)
+            {
+                return NotFound();
+            }
+
+            if (order.Status != OrderStatusCodes.Finalized && order.Status != OrderStatusCodes.Complete)
+            {
+                ErrorMessage = "Don't generate an email in this status.";
+                return RedirectToAction("Orders");
+            }
+
+            await _orderMessageService.EnqeueDisposalMessage(order);
+            var user = _dbContext.Users.Single(a => a.Id == CurrentUserId);
+            order.History.Add(new History
+            {
+                Action = "Disposal Waring Email",
+                Status = order.Status,
+                ActorId = user.NormalizedUserName,
+                ActorName = user.Name,
+                JsonDetails = order.JsonDetails,
+            });
+
+            await _dbContext.SaveChangesAsync();
+
+            Message = "Email Generated";
+
+            return RedirectToAction("MailQueue", "Admin", new {id = order.Id});
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> GenerateDisposalEmails(string ids)

```

Comment:
@jSylvestre these two are basically the same thing so either move the core logic into a private method or more likely just use GenerateDisposalEmails for everything even if it's just one they are generating